### DISCUSSION TITLE
Erorr run on python

### DISCUSSION THREAD

**Discussion by Moess23** [community/community#67399] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hello, I am new to programming and I am working on a code made and API interface, through which you enter your data so that he can enter it, so the specific site as a robot, for example, for me to speed up the payment process on my online store, so I got this code after trouble, but I do not know what errors it contains, because no matter how much I try, it does not work, either it means not found url 404,  Or I do not work at all, so is the code that performs the job required, and what are the errors in it

from selenium import webdriver    # import webdriver module
from selenium.webdriver.common.keys import Keys    # import keys module

def login_to_edfali(username: str, password: str) -> None:
    """
    Login to Edfali website.

    Parameters
    ----------
    username: str
        Username of edfali account.
    password: str
        Password of edfali account.

    Returns
    -------
    None

    """
    # إعداد متصفح الويب
    driver = webdriver.Firefox(
        executable_path="C:\\Users\\محمد\\AppData\\Local\\Programs\\Python\\Python311\\geckodriver.exe")    # initialize web driver

    # اذهب إلى صفحة تسجيل الدخول
    driver.get('https://edfali.bcd.ly/Edfali/Login.aspx')    # open url in browser

    # العثور على حقول الإدخال
    username_field = driver.find_element_by_name('Phone')    # find the element with name attribute value username
    password_field = driver.find_element_by_name('Pin')    # find the element with name attribute value password

    # إدخال بيانات الاعتماد
    username_field.send_keys(Phone)    # Send the value to username field
    password_field.send_keys(Pin)    # Send the value to password field

    # اضغط Enter لتسجيل الدخول
    password_field.send_keys(Keys.RETURN)    # Press Enter key

login_to_edfali('your-Phone', 'your-Pin')

**Comment by seyLu** [comment]
#### Hi @Moess23 

Things to note:
-  In your code, you're trying to send `Phone` and `Pin` as values for the `username` and `password` fields.

```diff
- username_field.send_keys(Phone)
- password_field.send_keys(Pin)
+ username_field.send_keys(username)
+ password_field.send_keys(password)
```

<br>

- make sure this is the correct path for `geckodriver.exe`:
> [!Warning]
C:\\Users\\محمد\\AppData\\Local\\Programs\\Python\\Python311\\geckodriver.exe

<br>

- if you are running this as a script, don't forget to add `if __name__ == '__main__'`

```diff
- login_to_edfali('your-Phone', 'your-Pin')
+ if __name__ == '__main__':
+     login_to_edfali('your-Phone', 'your-Pin')
```

<br>

---

<br>

```py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys


def login_to_edfali(username: str, password: str) -> None:
    """
    Login to Edfali website.

    Parameters
    ----------
    username: str
        Username of edfali account.
    password: str
        Password of edfali account.

    Returns
    -------
    None

    """
    # إعداد متصفح الويب
    driver = webdriver.Firefox(
        executable_path="C:\\Users\\محمد\\AppData\\Local\\Programs\\Python\\Python311\\geckodriver.exe")

    # اذهب إلى صفحة تسجيل الدخول
    driver.get('https://edfali.bcd.ly/Edfali/Login.aspx')

    # العثور على حقول الإدخال
    username_field = driver.find_element_by_name('Phone')
    password_field = driver.find_element_by_name('Pin')

    # إدخال بيانات الاعتماد
    username_field.send_keys(username)
    password_field.send_keys(password)

    # اضغط Enter لتسجيل الدخول
    password_field.send_keys(Keys.RETURN)



if __name__ == '__main__':
    # Usage
    login_to_edfali('your-Phone', 'your-Pin')
```

For easier debugging, you can check out python [logging](https://docs.python.org/3/library/logging.html) module, and add logging statements so you can trace what your code is doing.

It would also help to include the error logs, and the steps to reproduce the error, so devs can better assist you.

**Comment by mramitdas** [comment]
can you please share the error traceback?
